packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -619,8 +619,8 @@ abstract class Page<T> extends RouteSettings {
 /// An interface for observing the behavior of a [Navigator].
 class NavigatorObserver {
   /// The navigator that the observer is observing, if any.
-  NavigatorState? get navigator => _navigator;
-  NavigatorState? _navigator;
+  NavigatorState? get navigator => _navigators[this];
+  static final _navigators = Expando<NavigatorState>();
```

Comment:
(Also - as the analyzer check will hopefully remind you - this field needs to be typed)

Reply:
Done.